airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,545 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import copy
+import json
+import sys
+from datetime import datetime, timedelta, timezone
+from time import sleep
+from typing import TYPE_CHECKING, List, Optional, cast
+
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    from cached_property import cached_property
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.operators.python import ShortCircuitOperator
+from airflow.providers.amazon.aws.hooks.appflow import AppflowHook
+from airflow.providers.amazon.aws.utils import datetime_to_epoch_ms, get_airflow_version
+
+if TYPE_CHECKING:
+    from mypy_boto3_appflow.type_defs import (
+        DescribeFlowExecutionRecordsResponseTypeDef,
+        DescribeFlowResponseTypeDef,
+        ExecutionRecordTypeDef,
+        TaskTypeDef,
+    )
+
+    from airflow.utils.context import Context
+
+EVENTUAL_CONSISTENCY_OFFSET: int = 15  # seconds
+EVENTUAL_CONSISTENCY_POLLING: int = 10  # seconds
+SUPPORTED_SOURCES = {"salesforce", "zendesk"}
+
+
+class AppflowBaseOperator(BaseOperator):
+    """
+    Amazon Appflow Base Operator class (not supposed to be used directly in DAGs).
+
+    :param source: The source name (e.g. salesforce)
+    :param name: The flow name
+    :param source_field: The field name to apply filters
+    :param dt: The date value (or template) to be used in filters.
+    :param poll_interval: how often in seconds to check the query status
+    :param aws_conn_id: aws connection to use
+    :param region: aws region to use
+    """
+
+    BLUE = "#2bccbd"
+    ui_color = BLUE
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        flow_update: bool,
+        source_field: Optional[str] = None,
+        dt: Optional[str] = None,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        if source not in SUPPORTED_SOURCES:
+            raise AirflowException(f"{source} is not a supported source (options: {SUPPORTED_SOURCES})!")
```

Comment:
```suggestion
            raise ValueError(f"{source!r} is not a supported source (options: {SUPPORTED_SOURCES})!")
```
This exception seems more appropriate.

Reply:
Done!